local/local.go

```diff
@@ -110,67 +106,29 @@ func (i *Image) SetOS(osVal string) error {
 var emptyHistory = v1.History{Created: v1.Time{Time: imgutil.NormalizedDateTime}}
 
 func (i *Image) AddLayer(path string) error {
-	diffID, err := calculateChecksum(path)
-	if err != nil {
-		return err
-	}
-	layer, err := i.addLayerToStore(path, diffID)
+	layer, err := i.store.AddLayer(path)
 	if err != nil {
 		return err
 	}
 	return i.AddLayerWithHistory(layer, emptyHistory)
 }
 
-func calculateChecksum(path string) (string, error) {
```

Comment:
This function relies on the daemon always outputting uncompressed layers, which is no longer true